src/app/movie.service.ts

```diff
@@ -1,33 +1,44 @@
 import { Injectable } from '@angular/core';
 import { HttpClient } from '@angular/common/http';
 import { environment } from 'src/environments/environment';
+import { Observable, BehaviorSubject } from 'rxjs';
+import { mergeMap } from 'rxjs/operators';
 
 @Injectable({ providedIn: 'root' })
 
 export class MovieService {
+
+  private moviesChange: BehaviorSubject<void>;
   
   constructor(
     private http: HttpClient
-  ) { }
+  ) {
+    this.moviesChange = new BehaviorSubject(null);
+  }
 
-  public getAll(): Promise<Movie[]> {
-    return this.http.get<Movie[]>(`${environment.apiUrl}/movies`).toPromise();
+  public getAll(): Observable<Movie[]> {
```

Comment:
By using observables, we avoid having to do reloading in the list of movies page since we can automatically trigger a new fetch every time there's a change in movies (create, update, delete). We bind those change events to this fetch mechanism using the behaviour subject created in the constructor.